pkg/utils/addon_config.go

```diff
@@ -120,14 +120,15 @@ func GetAddOnDeploymentConfigSpecHash(config *addonapiv1alpha1.AddOnDeploymentCo
 	})
 }
 
-// GetAddOnConfigRef returns the first addon config ref for the given config type
+// GetAddOnConfigRef returns the last addon config ref for the given config type
 func GetAddOnConfigRef(
 	configReferences []addonapiv1alpha1.ConfigReference,
 	group, resource string) (bool, addonapiv1alpha1.ConfigReference) {
 
-	for _, config := range configReferences {
-		if config.Group == group && config.Resource == resource {
-			return true, config
+	// iterate the config reversely, the one with larger index is picked at first
```

Comment:
Have we already sorted this in another controller(with a tree structure)? I need to confirm this.

Reply:
this is use in the helm template addon to get values, which will need this to read in reverse order.